docs/source/guides/speed.rst

```diff
@@ -0,0 +1,301 @@
+.. testsetup:: *
+
+    from pytorch_lightning.trainer.trainer import Trainer
+    from pytorch_lightning.callbacks.early_stopping import EarlyStopping
+    from pytorch_lightning.core.lightning import LightningModule
+
+.. _speed:
+
+#######################
+Speed up model training
+#######################
+
+There are multiple ways you can speed up your model's time to convergence:
+
+* `<Early stopping_>`_
+
+* `<GPU/TPU training_>`_
+
+* `<Mixed precision (16-bit) training_>`_
+
+* `<Control Training Epochs_>`_
+* `<Control Validation Frequency_>`_
+
+* `<Limit Dataset Size_>`_
+
+.. _early_stopping:
+
+**************
+Early stopping
+**************
```

Comment:
there seems to be a misunderstanding. 
early stopping is a regularization technique. it does not decrease the time the model takes to converge. 

Here is a list of real ways one can increase convergence speed :) 
- changes to the optimization algorithm (different optimizer, different learning rate)
- different model hyperparameters (hidden size, number of layers)
- special normalization layers (batch norm, instance norm, etc.)
- different initialization

Reply:
Removed early stopping for now till we find the right place for it. 
Please feel free to add to the PR your suggestion for more ways to speed!